tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/files/BatchCopySpec.scala

```diff
@@ -105,24 +108,45 @@ class BatchCopySpec extends BaseIntegrationSpec {
     val uri         = s"/bulk/files/$destProjRef?storage=nxv:${destStorage.storageId}"
 
     for {
-      response   <- deltaClient.postAndReturn[Response](uri, payload, Coyote) { (json, response) =>
-                      (json, expectCreated(json, response))
-                    }
-      _          <- checkFileResourcesExist(destProjRef, response)
-      assertions <- checkFileContentsAreCopiedCorrectly(destProjRef, sourceFiles, response)
-    } yield assertions.head
+      response <- deltaClient.postAndReturn[Response](uri, payload, Coyote) { (json, response) =>
+                    (json, expectCreated(json, response))
+                  }
+      ids       = response.ids
+      _        <- checkFileResourcesExist(destProjRef, ids)
+      _        <- checkFileContentsAreCopiedCorrectly(destProjRef, sourceFiles, ids)
+      _        <- assertFileUserMetadataWasCopiedCorrectly(destProjRef, sourceFiles, ids)
+    } yield succeed
   }
 
-  def checkFileContentsAreCopiedCorrectly(destProjRef: String, sourceFiles: List[FileInput], response: Response) =
-    response.ids.zip(sourceFiles).traverse { case (destId, FileInput(_, filename, contentType, contents)) =>
+  def checkFileContentsAreCopiedCorrectly(destProjRef: String, sourceFiles: List[FileInput], ids: List[String]) =
+    ids.zip(sourceFiles).traverse { case (destId, FileInput(_, filename, contentType, contents, _)) =>
       deltaClient
         .get[ByteString](s"/files/$destProjRef/${UrlUtils.encode(destId)}", Coyote, acceptAll) {
           filesDsl.expectFileContentAndMetadata(filename, contentType, contents)
         }
     }
 
-  def checkFileResourcesExist(destProjRef: String, response: Response) =
-    response.ids.traverse { id =>
+  def assertFileUserMetadataWasCopiedCorrectly(
+      destProjRef: String,
+      sourceFiles: List[FileInput],
+      ids: List[String]
+  ): IO[Assertion] = {
+    ids
+      .zip(sourceFiles)
+      .parTraverse { case (id, FileInput(_, _, _, _, keywords)) =>
+        deltaClient.get[Json](s"/files/$destProjRef/${UrlUtils.encode(id)}", Coyote) { (json, response) =>
+          response.status shouldEqual StatusCodes.OK
+          json.hcursor.downField("keywords").as[Map[String, String]].toOption match {
+            case Some(value) => value shouldEqual keywords
+            case None        => fail("keywords missing")
+          }
```

Comment:
could do
```suggestion
          json.hcursor.get[Map[String, String]]("keywords").value shouldEqual keywords
```

Reply:
Yeah the failure message was not good when I had that, that's why I used this match

Reply:
ah yeah I've found that with these tests too, I quite like just `actual shouldEqual Right(expected)`